servers/src/main/java/tachyon/worker/lineage/LineageDataManager.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.lineage;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+import tachyon.Constants;
+import tachyon.Sessions;
+import tachyon.TachyonURI;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.InvalidStateException;
+import tachyon.exception.NotFoundException;
+import tachyon.thrift.FileDoesNotExistException;
+import tachyon.thrift.FileInfo;
+import tachyon.underfs.UnderFileSystem;
+import tachyon.util.io.PathUtils;
+import tachyon.worker.block.BlockDataManager;
+import tachyon.worker.block.io.BlockReader;
+
+/**
+ * Responsible for managing the lineage storing into under file system.
+ */
+public final class LineageDataManager {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+
+  UnderFileSystem mUfs;
+
+  /** Block data manager for access block info */
+  private final BlockDataManager mBlockDataManager;
+  private final List<Long> mPersistedFiles;
+  private final TachyonConf mTachyonConf;
+
+  public LineageDataManager(BlockDataManager blockDataManager, TachyonConf tachyonConf) {
+    mBlockDataManager = Preconditions.checkNotNull(blockDataManager);
+    mPersistedFiles = Lists.newArrayList();
+    mTachyonConf = Preconditions.checkNotNull(tachyonConf);
+    // Create Under FileSystem Client
+    String ufsAddress = mTachyonConf.get(Constants.UNDERFS_ADDRESS);
+    mUfs = UnderFileSystem.get(ufsAddress, mTachyonConf);
+  }
+
+  /**
+   * Persists the blocks of a file into the under file system.
+   *
+   * @param fileId the id of the file.
+   * @param blockIds the list of block ids.
+   * @throws IOException
+   */
+  public synchronized void persistFile(long fileId, List<Long> blockIds)
+      throws IOException {
+    String ufsDataFolder = mTachyonConf.get(Constants.UNDERFS_DATA_FOLDER);
```

Comment:
is this still true after #1550 ?

You may want to make the path calculation a util function shared between BlockWorker and LineageWorker